baseband/vdif/header.py

```diff
@@ -496,23 +499,30 @@ class VDIFSampleRateHeader(VDIFBaseHeader):
          ('sampling_rate', (4, 0, 23)),
          ('sync_pattern', (5, 0, 32, 0xACABFEED))))
 
-    # Add extra properties, ensuring 'time' comes after 'framerate', since
-    # time setting requires the frame rate.
+    # Add extra properties, ensuring 'time' comes after 'sample_rate', since
+    # time setting requires the sampling rate.
     _properties = (VDIFBaseHeader._properties[:-1] +
-                   ('bandwidth', 'framerate', 'time'))
+                   ('sample_rate', 'time'))
 
     def same_stream(self, other):
         return (super(VDIFSampleRateHeader, self).same_stream(other) and
                 self.words[4] == other.words[4] and
                 self.words[5] == other.words[5])
 
     @property
-    def bandwidth(self):
-        return u.Quantity(self['sampling_rate'],
-                          u.MHz if self['sampling_unit'] else u.kHz)
-
-    @bandwidth.setter
-    def bandwidth(self, bandwidth):
+    def sample_rate(self):
+        """Number of complete samples per second."""
+        return u.Quantity(self['sampling_rate'] *
+                          (1000000 if self['sampling_unit'] else 1000) *
```

Comment:
Can we keep my expression above, which sets the unit appropriately (rather than multiply by 1000000 or 1000)? That way the `repr` will more naturally look OK.

Also, looking at this again, I worry whether one should indeed divide by `nchan` (I know I implemented it as such in framerate but I am not sure this is properly tested anywhere... could you re-check the VDIF documentation?).

Reply:
Both EDV 1 and 3 specs say `sampling_rate` is a sample rate in ksps or Msps (so we shouldn't be dividing by `bps`, and we don't).  EDV 1's documentation compares the range of possible sampling rates to the K5/VSSP data acquisition systems (from 0.04 to 64 MHz).  A [K5 specification doc I found](ftp://ivscc.gsfc.nasa.gov/pub/TOW/tow2011/notebook/Kokado.Sem1.pdf) quotes sample rate values per channel (so the total data rate is `sample_rate * nchan * bps`), which suggests `sampling_rate` **shouldn't** be divided by `nchan` since it's already the per-channel rate.

However, for EDV 3, the convention is, "Each VDIF thread produced contains a single base-band channel", so `nchan = 1`.  EDV 1 doesn't make this specification, but it isn't explicit in what `sampling_rate` should be for multiple channels.

I'm worried this is actually an ambiguity in EDV 1 (the original VDIF specification doesn't have a sample rate, so that's okay).  Might pay to e-mail the NRAO folks.  In the meantime, this seems like the wrong PR to fix this issue (especially since it might break some other people's codes)?

Regardless, I've switched back to your expression above.

Reply:
Actually, on second thought it's the right PR.  I'd still like confirmation from an external source, though.  Let's discuss it tomorrow at one of the meetings.

Reply:
I am nearly sure we've encountered standard VDIF with single-thread, multiple channels and EDV=1, so I think the current implementation cannot be too wrong... Anyway, let's discuss outside of this PR -- goal now is simply to keep things consistent.